packages/docusaurus-utils/src/globUtils.ts

```diff
@@ -68,7 +69,9 @@ export function createAbsoluteFilePathMatcher(
 
   function getRelativeFilePath(absoluteFilePath: string) {
     const rootFolder = rootFolders.find((folderPath) =>
-      absoluteFilePath.startsWith(folderPath),
+      [addSuffix(folderPath, '/'), addSuffix(folderPath, '\\')].some((p) =>
+        absoluteFilePath.startsWith(p),
+      ),
```

Comment:
Hacky but this will work for everyone. Remember that a single config may be used to build sites on both Windows and Unix, so we need to cater to 2×2 = 4 cases. I would merge like this.